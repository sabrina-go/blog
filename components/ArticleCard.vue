<template>
  <NuxtLink
    class="article-card"
    :to="{ name: 'blog-slug', params: { slug: article.slug } }"
  >
    <h2 class="article-card__title">{{ article.title }}</h2>
    <p>{{ article.description }}</p>
  </NuxtLink>
</template>

<script>
export default {
  name: 'ArticleCard',
  props: {
    article: {
      type: Object,
      required: true,
    },
  },
}
</script>

<style lang="scss" scoped>
.article-card {
  background-color: $white;
  box-shadow: 0 10px 20px 0 #6b6880;
  border-radius: 16px;
  outline: none;
  color: $bright-gray;
  text-decoration: none;
  padding: 14px;

  &__title {
    font-family: $font-open-sans;
    font-weight: $font-semi-bold;
    font-size: 1.4rem;
    word-wrap: break-word;
  }
}
</style>
