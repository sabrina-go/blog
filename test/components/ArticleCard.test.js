import { shallowMount } from '@vue/test-utils';
import ArticleCard from '@/components/ArticleCard';

describe('components | ArticleCard', () => {
  const article = {
    uid: 'test',
    data: {
      title: [{ text: 'Article Card test' }],
      excerpt: 'description',
      icon: { data: { image: 'img' } },
    },
  };
  let component;

  beforeEach(() => {
    component = shallowMount(ArticleCard, {
      stubs: {
        NuxtLink: true,
        PrismicRichText: true,
        PrismicImage: true,
      },
      propsData: {
        article,
      },
    });
  });

  test('is a Vue instance', () => {
    expect(component.vm).toBeTruthy();
  });

  test('render properly', () => {
    expect(component.html()).toMatchSnapshot();
  });

  test('should create one nuxt-link', () => {
    expect(component.findAll('nuxtlink-stub').length).toBe(1);
  });

  test('should use article title', () => {
    expect(component.find('h2').text()).toBe(article.data.title[0].text);
  });
});
