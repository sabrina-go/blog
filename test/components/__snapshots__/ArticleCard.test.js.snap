// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | ArticleCard render properly 1`] = `
"<nuxtlink-stub to=\\"[object Object]\\" class=\\"article-card\\">
  <h2 class=\\"article-card__title\\">Article Card test</h2>
  <p>description</p>
  <div class=\\"article-card__icon\\"><img role=\\"presentation\\" alt=\\"\\"></div>
</nuxtlink-stub>"
`;
